### PR TITLE
New sharing_disabled backfill script

### PR DESCRIPTION
**Pull request by islemaster**
Uses [`activerecord-import`](https://github.com/zdennis/activerecord-import) to rapidly update a value in the `properties` JSON blob in our `users` table for roughly 35 million users, in batches of 10,000.  Will ran some tests and found that this approach can update 10k rows in about 1.6 seconds so we're estimating a little over 90 minutes to run this query (against 97% of our user rows!).

## Next steps
We need to deploy this change so that the `activerecord-import` gem gets installed on production-console, then we'll manually run the oneoff script to update user rows.

## Why
We're going with this approach (updating so many rows) instead of some other approaches we discussed where application logic is used to minimize the number of rows we need to update, because:
- The application implementation for this approach is already done.
- We've got a small number of user rows that have already been updated from a previous attempt (no user bad, but it'd be nice to be consistent again)
- There are product implications of other approaches we discussed, and we know this approach meets spec.

## Awesomeness
@wjordan wrote:

> I'd like to see us start integrating `activerecord-import` into other parts of our stack, specifically bulk seeding steps.

We can't wait to see what sort of other speedups this permits...

### CODE REVIEW

**Comment by Erin007** [comment]
Do we still have to consider validation, like in @caleybrock 's earlier PR #21671?

**Comment by islemaster** [comment]
I believe the `validate: false` on [this line](https://github.com/code-dot-org/code-dot-org/pull/21681/files#diff-0f119836895409b54f02aac4f8473916R16) deals with that.

File: dashboard/scripts/update_under_13_students.rb [hunk 1]
```diff
@@ -6,11 +6,16 @@
 puts "Starting to batch update all students under 13."
 num_students_updated = 0
 
+batch_size = 10000
 min_birthday = Date.today - 13.years
-User.where('birthday IS NULL OR birthday > ?', min_birthday).find_each do |user|
-  user.update! sharing_disabled: true
-  num_students_updated += 1
-  puts "Updated #{num_students_updated} students so far." if num_students_updated % 100000 == 0
+User.where('birthday IS NULL OR birthday > ?', min_birthday).in_batches(of: batch_size) do |where|
```
Comment (balderdash) [inline dashboard/scripts/update_under_13_students.rb:11]: Seems like the variable should be named something like `users` instead of `where`.

**Comment by ewjordan** [comment]
This is awesome, I'm *definitely* going to use this approach next time I have to do a big data migration. Good find!

File: dashboard/scripts/update_under_13_students.rb (hunk 1 above)
Comment (aoby) [reply dashboard/scripts/update_under_13_students.rb:11]: or `batch`

File: dashboard/scripts/update_under_13_students.rb [hunk 2]
```diff
@@ -6,11 +6,16 @@
...
+  values = where.pluck(:id, :properties)
+  values.each do |_id, properties|
+    properties['sharing_disabled'] = true
+  end
+  User.import([:id, :properties], values, validate: false, on_duplicate_key_update: [:properties])
```
Comment (joshlory) [inline dashboard/scripts/update_under_13_students.rb:16]: @islemaster what does `on_duplicate_key_update` do here?

Comment (islemaster) [reply dashboard/scripts/update_under_13_students.rb:16]: The default mode for the `activerecord-import` gem is a bulk-insert, but here we want to do a bulk-update.  The `on_duplicate_key_update` setting turns this into an upsert operation that will update the `properties` column when primary keys match (in this case `id`).

See also: [On Duplicate Key Update on the activerecord-import wiki](https://github.com/zdennis/activerecord-import/wiki/On-Duplicate-Key-Update)

Comment (aoby) [reply dashboard/scripts/update_under_13_students.rb:16]: @joshlory See https://github.com/zdennis/activerecord-import/wiki/On-Duplicate-Key-Update

TL;DR - it's an upsert, when the key (id) already exists update properties
